tables/automl/notebooks/purchase_prediction/purchase_prediction.ipynb

```diff
@@ -1,909 +1,1242 @@
-        "create_model_result = create_model_response.result()\n",
-        "model_name = create_model_result.name\n",
-        "print(model_name)"
-      ],
-      "execution_count": 0,
-      "outputs": []
-    },
-    {
-      "metadata": {
-        "id": "tCIk1e4UuDxZ",
-        "colab_type": "text"
-      },
-      "cell_type": "markdown",
-      "source": [
-        "# 6. Make a prediction"
-      ]
-    },
-    {
-      "metadata": {
-        "id": "H7Fi5f9zuG5f",
-        "colab_type": "text"
-      },
-      "cell_type": "markdown",
-      "source": [
-        "There are two different prediction modes: online and batch. The following cell shows you how to make a batch prediction."
-      ]
-    },
-    {
-      "metadata": {
-        "id": "AZ_CPff77m4e",
-        "colab_type": "code",
-        "cellView": "both",
-        "colab": {}
-      },
-      "cell_type": "code",
-      "source": [
-        "#@title Start batch prediction { vertical-output: true, output-height: 200 }\n",
-        "print(client.list_models(location_path))\n",
-        "\n",
-        "batch_predict_gcs_input_uris = ['gs://trial_for_c4m/validation_unnested_FULL.csv',] #@param\n",
-        "batch_predict_gcs_output_uri_prefix = 'gs://trial_for_c4m' #@param {type:'string'}\n",
-        "# Define input source.\n",
-        "batch_prediction_input_source = {\n",
-        "  'gcs_source': {\n",
-        "    'input_uris': batch_predict_gcs_input_uris\n",
-        "  }\n",
-        "}\n",
-        "# Define output target.\n",
-        "batch_prediction_output_target = {\n",
-        "    'gcs_destination': {\n",
-        "      'output_uri_prefix': batch_predict_gcs_output_uri_prefix\n",
-        "    }\n",
-        "}\n",
-        "batch_predict_response = prediction_client.batch_predict(\n",
-        "    model_name, batch_prediction_input_source, batch_prediction_output_target)\n",
-        "print('Batch prediction operation: {}'.format(batch_predict_response.operation))\n",
-        "# Wait until batch prediction is done.\n",
-        "batch_predict_result = batch_predict_response.result()\n",
-        "batch_predict_response.metadata\n",
-        "\n"
-      ],
-      "execution_count": 0,
-      "outputs": []
-    },
-    {
-      "metadata": {
-        "id": "utGPmXI-uKNr",
-        "colab_type": "text"
-      },
-      "cell_type": "markdown",
-      "source": [
-        "#7. Evaluate your prediction"
-      ]
+    "colab_type": "code",
+    "id": "KIdmobtSsPj8",
+    "outputId": "14c234ca-5070-4301-a48c-c69d16ae4c31"
+   },
+   "outputs": [],
+   "source": [
+    "%pip install --quiet google-cloud-automl"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import libraries and define constants\n",
+    "\n",
+    "First, import Python libraries required for training,\n",
+    "The code example below demonstrates importing the AutoML Python API module into a python script. "
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
this alignment looks odd, I'm unfamiliar with conventions advocating for 3 spaces.